fuse/daemon.cpp

```diff
@@ -1525,6 +1532,10 @@ int main(int argc, char *argv[])
 	fl.l_len = 0; // 0=largest possible
 	if (fcntl(log, F_SETLK, &fl) != 0) {
 		if (errno == EAGAIN || errno == EACCES) {
+			if (enable_trace) {
+				fprintf(stderr, "fcntl(%s.log): %s -- assuming another daemon exists\n",
+					path.c_str(), strerror(errno));
+			}
 			status = 0; // another daemon is already running
```

Comment:
Is status = 0 the same as when it starts up a new daemon, it could be nice to have a separate return code so that the client can know what happened

Reply:
Could check it in the `do waitpid(pid, &status, 0);`

Reply:
Unfortunately, by that point the daemon already ran fork twice to cut its parent-client relationship with the client, so the client cannot recover the exit status using waitpid.

Reply:
The 'waitpid' call in the client is just there to reap the exit status of the pre-fork daemon child so it doesn't pollute the process tree.